larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewSimpleTracksTool.cc

```diff
@@ -95,13 +88,16 @@ bool TwoViewSimpleTracksTool::PassesElementCuts(MatrixType::ElementList::const_r
 
     const TwoViewXOverlap &xOverlap(eIter->GetOverlapResult().GetTwoViewXOverlap());
 
-    if ((xOverlap.GetXSpan0() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction0() > m_minXOverlapFraction) &&
-        (xOverlap.GetXSpan1() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction1() > m_minXOverlapFraction))
+    if (!((xOverlap.GetXOverlapFraction0() > m_minXOverlapFraction) &&
+         (xOverlap.GetXOverlapFraction1() > m_minXOverlapFraction)))
     {
-        return true;
+        return false;
     }
 
-    return false;
+    if (!eIter->GetOverlapResult().IsInitialized())
```

Comment:
This check should go first, at the top of this function, otherwise I think some of the function calls above, e.g. GetXOverlapFraction0(), etc. might raise an exception.